src/app/ui/player-quest/player-quest.service.ts

```diff
@@ -226,7 +234,17 @@ export class PlayerQuestService {
       });
     });
 
-    return from(trans);
+    return from(trans).pipe(
+      flatMap(() => eligibleForGoodWorkBadge ? this.awardGoodWorkBadge(quest) : of(null))
```

Comment:
It would be nicer to pass eligibleForGoodWorkBadge as a parameter and deal with it inside the award function
```suggestion
      flatMap(() => this.awardGoodWorkBadge(quest, eligibleForGoodWorkBadge))
```

Reply:
actually, awardGoodWorkBadge should be moved to badges.service instead